src/gt4py/backend/module_generator.py

```diff
@@ -78,24 +81,23 @@ def make_args_data_from_iir(implementation_ir: gt_ir.StencilImplementation) -> M
                 access |= AccessKind.READ
             if arg.name in out_fields:
                 access |= AccessKind.WRITE
-            if arg.name not in implementation_ir.unreferenced:
-                field_decl = implementation_ir.fields[arg.name]
-                data.field_info[arg.name] = FieldInfo(
-                    access=access,
-                    boundary=implementation_ir.fields_extents[arg.name].to_boundary(),
-                    axes=tuple(field_decl.axes),
-                    data_dims=tuple(field_decl.data_dims),
-                    dtype=field_decl.data_type.dtype,
-                )
-            else:
-                data.field_info[arg.name] = None
+            field_decl = implementation_ir.fields[arg.name]
+            data.field_info[arg.name] = FieldInfo(
+                access=access,
+                boundary=implementation_ir.fields_extents[arg.name].to_boundary(),
+                axes=tuple(field_decl.axes),
+                data_dims=tuple(field_decl.data_dims),
+                dtype=field_decl.data_type.dtype,
+            )
         else:
-            if arg.name not in implementation_ir.unreferenced:
-                data.parameter_info[arg.name] = ParameterInfo(
-                    dtype=implementation_ir.parameters[arg.name].data_type.dtype
-                )
-            else:
-                data.parameter_info[arg.name] = None
+            data.parameter_info[arg.name] = ParameterInfo(
+                access=(
+                    AccessKind.NONE
+                    if arg.name in implementation_ir.unreferenced
```

Comment:
Since you are not using `implementation_ir.unreferenced` in the above branch anymore I would suggest to also remove the dependency on it here:
```suggestion
                    if arg.name not in in_fields and arg.name not in out_fields
```